Sources/Scheduler.swift

```diff
@@ -13,6 +13,48 @@ import Foundation
 	import CDispatch
 #endif
 
+extension Date {
+	internal func addingTimeInterval(_ interval: DispatchTimeInterval) -> Date {
+		return addingTimeInterval(interval.asTimeInterval())
+	}
+}
+
+extension DispatchTimeInterval {
+	internal func asTimeInterval() -> TimeInterval {
+		switch self {
+		case let .seconds(s):
+			return TimeInterval(s)
+		case let .milliseconds(ms):
+			return TimeInterval(TimeInterval(ms) / 1000.0)
+		case let .microseconds(us):
+			return TimeInterval( UInt64(us) * NSEC_PER_USEC ) / TimeInterval(NSEC_PER_SEC)
+		case let .nanoseconds(ns):
+			return TimeInterval(ns) / TimeInterval(NSEC_PER_SEC)
+		}
+	}
+
+	internal static prefix func -(lhs: DispatchTimeInterval) -> DispatchTimeInterval {
+		switch lhs {
+		case let .seconds(s):
+			return .seconds(-s)
+		case let .milliseconds(ms):
+			return .milliseconds(-ms)
+		case let .microseconds(us):
+			return .microseconds(-us)
+		case let .nanoseconds(ns):
+			return .nanoseconds(-ns)
+		}
+	}
+
+	/// Scales a time interval by the given scalar specified in `rhs`.
+	///
+	/// - returns: Scaled interval in microseconds
+	internal static func *(lhs: DispatchTimeInterval, rhs: Double) -> DispatchTimeInterval {
+		let seconds = lhs.asTimeInterval() * rhs
+		return .microseconds(Int(seconds * 1000 * 1000))
```

Comment:
@NachoSoto So this is *only* used internally to "scale down" a time interval. Specifically it's used only to scale intervals to 10% of their original value for the default leeway parameter.

But you are correct. If seconds is over ~200,000, 10% is 2000, and hence we end up with ~2billion which will overflow a signed integer *on 32-bit platforms* (of which there aren't really any more for much longer, anyway…)

But I'm pretty sure that even 200k seconds is going to be a rarely (if ever) specified interval for our APIs.

One of the tests I plan to add is to find this tipping point.